examples/scene/resources/scripts/melee.lua

```diff
@@ -2,48 +2,65 @@ include("resources:scripts/common")
 
 melee = room {
     background = "resources:backgrounds/Melee",
-    enter = function(self)
-        local pirate1_dialog_props = { pos = {x=60, y=20}, color = magenta }
-        local pirate2_dialog_props = { pos = {x=60, y=50}, color = yellow }
-
-        guybrush:show{
-            pos={x=340, y=140}, 
-            dir=left,
-            costume="resources:costumes/Guybrush" -- TODO: this must be a variable
+    objects = {
+        bucket = {
+            name = "bucket",
+            sprites = "resources:sprites/objects",
+            pos = {x=300, y=180},
+            hotspot = {x=290, y=190, w=20, h=20},
+            usedir = left,
+            usepos = {x=290, y=190},
+            states = {
+                default = {
+                    anim = { index = 0, frames = 0, delay = 0 },
+                },
+                pickup = {}
```

Comment:
What is this?

Reply:
Do you mean the `pickup` state? It is the object state once it is picked up. 

In Scumm, whether the object is rendered in the room or not is managed with states. One state corresponds to the object in the room, the other out of the room (in someone's inventory). 

This state is assuming the same approach. But I am still not sure if it is the way to go. Perhaps we can check the object owner and render the object only when it is nil instead of having an special state for it.